### ISSUE TITLE
[WOTLK] Corpse Decay Rate Locked Into Corpse Looted Decay Rate

### ISSUE THREAD

**Issue by Vladskywolf** [cmangos/issues#990]
**Current behavior**: 
Normal mobs which have not been looted are decaying after approx 95 seconds. This does not change even after adjusting parameter Corpse.Decay.NORMAL in mangosd.conf from 300 to 900. According to parameter descriptions, mobs seem to decay after Corpse.Decay.NORMAL \* Rate.Corpse.Decay.Looted (1/3 of the normal time when set to 0.0).

**Expected behavior**: 
Normal unlooted mobs should not decay until 300 seconds have passed.

**Steps to reproduce**:
1. Kill a normal mob
2. Stand, watch and have a stopwatch ready. :)

**Client version**: 
3.3.5 [12340]

**Commit hash**: 
3af38d5d2e955492fbc13c4db5aa0ba8689a36d3

**Database version**: 
UDB 0.12.2 updatepack 408 (with updates to 495)

**Operating system**: 
Ubuntu 12.04

Tested decay rate in v18 build 12938 on saved virtual machine snapshot. Mob decayed after 240 seconds. Still faster than normal but better. Something has obviously changed since then.

**Comment by Nightfly510** [comment]
Have you checked the respawn time for these creatures in the database?  Pretty sure if creature respawn < corpse decay, respawn time wins out.  Also, according to how I read config file, using the default value of 0.0 for Rate.Corpse.Decay.Looted results in it using 1/3 the _respawn time_, not corpse decay time.  But this may be a typo... I should check the code to be sure.

Yep, just verified: a value of 0.0 uses 1/3 of the _respawn time_.

You probably just want to change this to near 1.0 to use the full corpse decay time, and may want to adjust respawn times as well.

**Comment by Vladskywolf** [comment]
I tested the same creature, Twilight Thug (2339) with a spawntimesecs value of 275, on both versions, looted and unlooted. 

In build 12938, the non-looted twilight thug decays after approx. 240 seconds, probably in preparation for respawning in 275 seconds as you pointed out (thanks for that, wasn't aware of it).

In build 12940 (commit hash stated in first post), the non-looted twilight thug decays after approx. 95 seconds, despite a 275 second spawntimesecs value in the database. This is equal to the time it should take to decay _if it has been looted_, which in this case it has not been.

So there is definitely a problem. With few mobs in a fight, I didn't notice it. But when grouped and taking on multiple mobs in longer fights, often they'd start disappearing before we finished off the last mob. And no, the fights wouldn't last for 3-5 minutes, but sometimes for longer than a minute and a half. That's what caught my attention.

As far as the respawn rate after it's been looted, it really wasn't my main point, I only mentioned it because I saw a correlation between how this value was supposed to work, and what I was seeing for non-looted mobs.
